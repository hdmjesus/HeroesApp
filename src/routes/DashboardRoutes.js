import React from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import { NavBar } from "../Components/ui/NavBar";
import { DcScreen } from "../Screen/DcScreen";
import { HeroesScreen } from "../Screen/HeroesScreen";
import { MarvelScreen } from "../Screen/MarvelScreen";

export const DashboardRoutes = () => {
  return (
    <>
      <NavBar />
      <div>
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/heroe/:heroid" component={HeroesScreen} />
          <Route exact path="/dc" component={DcScreen} />
          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
