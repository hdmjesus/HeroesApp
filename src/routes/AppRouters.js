import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import { LoginScreen } from "../Screen/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";

const App = () => {
  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/login" component={LoginScreen} />
        <Route path="/" component={DashboardRoutes} />
      </Switch>
    </BrowserRouter>
  );
};

export default App;
