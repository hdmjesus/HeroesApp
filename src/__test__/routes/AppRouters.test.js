import React from "react";
import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import { AuthContext } from "../../Context/AuthContext";
import { AppRouters } from "../../routes/AppRouters";

describe("Pruebas en el componente <AppRouters/>", () => {
  test("should show login if is not authenticated ", () => {
    const { Provider } = AuthContext;
    const contextValue = { dispatch: jest.fn(), user: { logged: false } };
    const wrapper = mount(
      <Provider value={contextValue}>
        <AppRouters />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should show the ScreenMarvel if is authenticated ", () => {
    const { Provider } = AuthContext;
    const contextValue2 = {
      dispatch: jest.fn(),
      user: { logged: true, name: "Jesus Hernandez" },
    };
    const wrapper2 = mount(
      <Provider value={contextValue2}>
        <MemoryRouter>
          <AppRouters />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper2).toMatchSnapshot();
    console.log(wrapper2.html());
    expect(wrapper2.find(".navbar").exists()).toBe(true);
  });
});
