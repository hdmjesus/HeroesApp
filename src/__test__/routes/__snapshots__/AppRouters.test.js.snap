// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AppRouters/> should show login if is not authenticated  1`] = `
<div
  className="container mt-5"
>
  <h1
    className="m-3"
  >
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary m-4"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`Pruebas en el componente <AppRouters/> should show the ScreenMarvel if is authenticated  1`] = `
<div
  className="container mt-5"
>
  <h1
    className="m-3"
  >
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary m-4"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
